### PR TITLE
[refactor] Update test examples

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

This PR adds the examples to be run within subprocesses. 
I have observed the examples could hang which doesn't happen while running within a subprocess.

Fixes #\<issue_number>

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by carmocca** [comment]
This might not be necessary after the examples changes in https://github.com/PyTorchLightning/pytorch-lightning/commit/0dfc6a18bd0887261d238819cfaba6a54e6fe4e2

**Comment by Tshimanga** [comment]
@carmocca potentially... these test are still failing on my PR #9161 after pulling latest from master, but now instead of timing out it fails immediately. some assistance/input there would be great because I lack familiarity with what could be the issue. Doesn't seem like something related to the parts of the code that I touched

**Comment by carmocca** [comment]
> but now instead of timing out it fails immediately. some assistance/input there would be great because I lack familiarity with what could be the issue. Doesn't seem like something related to the parts of the code that I touched

When it fails immediately it's usually due to lack of resources on the GPU pool. I've re-run the job.

**Review by Borda — APPROVED** [review]

**Comment by tchaton** [comment]
> > but now instead of timing out it fails immediately. some assistance/input there would be great because I lack familiarity with what could be the issue. Doesn't seem like something related to the parts of the code that I touched
> 
> When it fails immediately it's usually due to lack of resources on the GPU pool. I've re-run the job.

I think we have an issue with dp on master. The examples seems to hang.

Best,
T.C

File: pl_examples/test_examples.py [hunk 1]
```diff
@@ -28,15 +35,81 @@
...
+
+
+@pytest.mark.parametrize("cli_args", [ARGS_DEFAULT])
+def test_examples_cpu_backbone_image_classifier(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.backbone_image_classifier", cli_args)
+
+
+@pytest.mark.parametrize("cli_args", [ARGS_DEFAULT])
+def test_examples_cpu_autoencoder(tmpdir, cli_args):
+    run(tmpdir, "pl_examples.basic_examples.autoencoder", cli_args)
```
Comment (carmocca) [inline pl_examples/test_examples.py:98]: This PR not necessary anymore - these are already run here

https://github.com/PyTorchLightning/pytorch-lightning/blob/91ce0d0a99ec0a488e04e09997f6dd662ce217cd/.azure-pipelines/gpu-tests.yml#L108-L112

Perhaps with the exception of the DALI example, but it doesn't seem to be failing, so why change it

**Review by carmocca — CHANGES_REQUESTED** [review]

**Pull request closed without merging**